packages/@lwc/synthetic-shadow/src/faux-shadow/element.ts

```diff
@@ -92,7 +92,11 @@ function shadowRootGetterPatched(this: Element): ShadowRoot | null {
 
 function childrenGetterPatched(this: Element): HTMLCollectionOf<Element> {
     const owner = getNodeOwner(this);
-    const childNodes = isNull(owner) ? [] : getAllMatches(owner, getFilteredChildNodes(this));
+    const filteredChildNodes = getFilteredChildNodes(this);
+    // no need to filter nodes by owner in case of light DOM
```

Comment:
This also applies to root shadow DOM components:
```suggestion
    // No need to filter by owner for non-shadowed elements
```